app/src/org/commcare/activities/InstallFromListActivity.java

```diff
@@ -422,4 +414,22 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    private void loadPreviouslyRetrievedAvailableApps() {
+        for (AppAvailableToInstall availableApp : storage()) {
```

Comment:
Wow this is cool! Embarrassing, I never thought to use an inline iterator with storage() before.

Will this throw a NullPointerException if storage() has no records?